test/catalog/test_catalog_manager.py

```diff
@@ -26,8 +26,7 @@ class CatalogManagerTests(unittest.TestCase):
     def __init__(self, *args, **kwargs):
         super().__init__(*args, **kwargs)
 
-    @mock.patch('src.catalog.catalog_manager.init_db')
-    def test_catalog_manager_singleton_pattern(self, mocked_db):
+    def test_catalog_manager_singleton_pattern(self):
```

Comment:
CatalogManager will be reused by other testcases. So init_db has to be called here.

Reply:
Also considering removing the rest of @mock.path in this file. Since once the catalogmanaer is created, the stored instance will be directly returned and it will not be initialized again.